### PR TITLE
#415: API-alphabetical from md to rst

### PR DESCRIPTION
**Pull request by antoinemeyer5**
## Notes
- This PR is part of #378
- I have tested all the links.
- The file is identical to the original.

## Screenshots
image
image
image
image
image
image
image

### CODE REVIEW

**Comment by fnrizzi** [comment]
> ine 12 and line 14: These are exactly the same lines. I have to remove the duplication, isn't it

yes

File: docs/source/API/alphabetical.rst [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+| `SpaceAccessibility <core/SpaceAccessibility.html>`_                                 | `Core <core-index.html>`_ | `Spaces <core/Spaces.html>`_                                           | Facility to query accessibility rules between execution and memory spaces.                                                              |
++--------------------------------------------------------------------------------------+---------------------------+------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
+| `Subview Type <core/view/Subview_type.html>`_                                        | `Core <core-index.html>`_ | `View and related <core/View.html>`_                                   | Type of multi-dimensional array which is returned by the subview function                                                               |
```
Comment (crtrott) [inline docs/source/API/alphabetical.rst:204]: We can't do this: the keyword is `Subview` as stupid as it is.

**Review by crtrott — CHANGES_REQUESTED** [review]
I don't think we can do the Subview thing this way. Otherwise looks good.

File: docs/source/API/alphabetical.rst (hunk 1 above)
Comment (fnrizzi) [reply docs/source/API/alphabetical.rst:204]: looking into how to get rid of "Type" but avoid warning

File: docs/source/API/alphabetical.rst [hunk 2]
```diff
@@ -54,6 +54,9 @@ Containers
 Core
 ----
 
+.. |SubviewType| replace:: Subview
+.. _SubviewType: core/view/Subview_type.html
```
Comment (fnrizzi) [inline docs/source/API/alphabetical.rst:58]: please write acomment to explain why this fixes it and your thinking behind this solution

**Comment by antoinemeyer5** [comment]
There are two ways to create internal links with rst:

(1) simple link: ``` `Subview <core/view/Subview_type.html>`_ ```

The first solution (1) is simple, but **makes no difference between the link name and its display**. Internally, the link is called *Subview* and the display clickable by the user is *Subview*. In our case, there were several links with the same name, so we had a duplication warning.

(2) more customizable link:
```rst
.. declaration
.. |SubviewType| replace:: Subview
.. _SubviewType: core/view/Subview_type.html

.. use
|SubviewType|_
```

This second solution (2) **does the same thing and display**, but differentiates between:
- the internal name of the link (here *SubviewType*),
- the display of the link to the user (here *Subview*) and
- the destination of the link.

Thanks to this, you can converse with the desired display and avoid internal duplication. You must always be careful not to have another link in this file whose name is *SubviewType*, otherwise the duplication warning will return.

With this solution, it's much simpler to have internal link names with a certain shape and a different display. This separates rst's internal logic from the user's display.

**Comment by fnrizzi** [comment]
ok makes sense to me

**Review by fnrizzi — APPROVED** [review]

**Comment by nmm0** [comment]
BTW - In the future, we could potentially rely on the [index](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#special-names) since it's auto-generated

**Comment by fnrizzi** [comment]
> BTW - In the future, we could potentially rely on the [index](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#special-names) since it's auto-generated

sounds a good idea

**Review by masterleinad — APPROVED** [review]

**Review by crtrott — APPROVED** [review]

**Pull request merged**